qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -116,39 +136,59 @@ public CompletableFuture<List<CodeAction>> doCodeActions(Template template, Code
 				QuteErrorCode errorCode = QuteErrorCode.getErrorCode(diagnostic.getCode());
 				if (errorCode != null) {
 					switch (errorCode) {
-						case UndefinedObject:
-							// The following Qute template:
-							// {undefinedObject}
-							//
-							// will provide a quickfix like:
-							//
-							// Declare `undefinedObject` with parameter declaration."
-							doCodeActionsForUndefinedObject(template, diagnostic, errorCode, codeActions);
-							break;
-						case UndefinedSectionTag:
-							// The following Qute template:
-							// {#undefinedTag }
-							//
-							// will provide a quickfix like:
-							//
-							// Create `undefinedTag`"
-							doCodeActionsForUndefinedSectionTag(template, diagnostic, codeActions);
-							break;
-						case UndefinedNamespace:
-							// The following Qute template:
-							// {undefinedNamespace:xyz}
-							doCodeActionToSetIgnoreSeverity(template, Collections.singletonList(diagnostic), errorCode,
-									codeActions, UNDEFINED_NAMESPACE_SEVERITY_SETTING);
-							break;
-						case UnknownProperty:
-							doCodeActionToCreateProperty(template, diagnostic, errorCode, codeActions,
-									codeActionResolveFutures,
-									resolver,
-									sharedSettings);
-							break;
-						default:
-							break;
+					case UndefinedObject:
+						// The following Qute template:
+						// {undefinedObject}
+						//
+						// will provide a quickfix like:
+						//
+						// Declare `undefinedObject` with parameter declaration."
+						doCodeActionsForUndefinedObject(template, diagnostic, errorCode, codeActions);
+						break;
+					case UndefinedSectionTag:
+						// The following Qute template:
+						// {#undefinedTag }
+						//
+						// will provide a quickfix like:
+						//
+						// Create `undefinedTag`"
+						doCodeActionsForUndefinedSectionTag(template, diagnostic, codeActions);
+						break;
+					case UndefinedNamespace:
+						// The following Qute template:
+						// {undefinedNamespace:xyz}
+						doCodeActionToSetIgnoreSeverity(template, Collections.singletonList(diagnostic), errorCode,
+								codeActions, UNDEFINED_NAMESPACE_SEVERITY_SETTING);
+						break;
+					case UnknownMethod:
+						// CodeAction(s) to replace text with similar suggestions
+						try {
+							String varName = QutePositionUtility
+									.findBestNode(template.offsetAt(diagnostic.getRange().getStart()), template
+											.findNodeBefore(template.offsetAt(diagnostic.getRange().getStart())))
+									.getNodeName();
+							doCodeActionsForSimilarValues(template, diagnostic, codeActions, varName);
+						} catch (BadLocationException e) {
+						}
+						break;
+					case UnknownProperty:
+						doCodeActionToCreateProperty(template, diagnostic, errorCode, codeActions,

```

Comment:
I will create doCodeActionForUnknownProperty and move doCodeActionToCreateProperty in thi snew method and manage your similar value.